monai/deploy/operators/monai_bundle_inference_operator.py

```diff
@@ -477,14 +485,19 @@ def compute(self, op_input: InputContext, op_output: OutputContext, context: Exe
 
             start = time.time()
             for name in self._inputs.keys():
-                value, metadata = self._receive_input(name, op_input, context)
+                # Input MetaTensor creation is based on the same logic in monai LoadImage
+                # value: NdarrayOrTensor  # MyPy complaints
+                value, meta_data = self._receive_input(name, op_input, context)
+                value = convert_to_dst_type(value, dst=value)[0]
+                if not isinstance(meta_data, dict):
+                    raise ValueError("`meta_data` must be a dict.")
+                value = MetaTensor.ensure_torch_and_prune_meta(value, meta_data)
                 inputs[name] = value
-                if metadata:
-                    inputs[(f"{name}_{self._meta_key_postfix}")] = metadata
+                # Named metadata dict not needed any more, as it is in the MetaTensor
 
             inputs = self.pre_process(inputs)
-            first_input = inputs.pop(first_input_name)[None].to(self._device)  # select first input
-            input_metadata = inputs.get(f"{first_input_name}_{self._meta_key_postfix}", None)
+            first_input_v = inputs[first_input_name]  # keep a copy of value for later use
```

Comment:
Do we need `deepcopy` here?

Thanks.

Reply:
The InvertD has changed to look for `image` dict name instead of `image_metadata_dict`. It was not clear to me if it matters the [`image`] value is before or post inference or even preprocessing. So, the intend here is really just to keep a reference to the value part of dict for adding back the key, `image`, because of the `pop` of key, `image`. I tried not to use `pop` but somehow direct using `inputs[first_input_name].to(self._device)` did not work.